src/sha256.rs

```diff
@@ -209,6 +209,20 @@ macro_rules! round(
 );
 
 impl HashEngine {
+    /// Create a new [HashEngine] from a midstate.
+    ///
+    /// Be aware that this method only works when [length] is a multiple of the block size.
```

Comment:
Can you `assert!` this? Ok if you add it in a separate commit @stevenroose

Reply:
Done.